tools/close_by_signature.py

```diff
@@ -0,0 +1,243 @@
+Root issues can be linked to given signature messages, and if provided they 
+will be linked to the resolved issues on close. 
+"""
+    parser = argparse.ArgumentParser(description=desc)
+    parser.add_argument(
+        '-v', '--verbose',
+        action='store_true',
+        help='Output verbose logging'
+    )
+
+    login_group = parser.add_argument_group(title='Login options')
+    login_args = login_group.add_mutually_exclusive_group()
+    login_args.add_argument(
+        '--netrc',
+        default='~/.netrc', 
+        required=False,
+        help='netrc file',
+    )
+    login_args.add_argument(
+        '-up', '--user-password', 
+        required=False,
+        help='Username and password in the format of user:pass',
+    )
+    
+    selectors_group = parser.add_argument_group(title='Issues selection')
+    selectors = selectors_group.add_mutually_exclusive_group(required=True)
+    selectors.add_argument(
+        '-r', '--recent-issues',
+        action='store_true',
+        help='Handle recent (30 days) Triaging Tickets',
+    )
+    selectors.add_argument(
+        '-a', '--all-issues',
+        action='store_true',
+        help='Handle all Triaging Tickets',
+    )
+    selectors.add_argument(
+        '-i', '--issue',
+        required=False,
+        help='Triage issue key',
+    )
+
+    filter_group = parser.add_argument_group(title='Filter by')
+    filter_group.add_argument(
+        '-st', '--signature-type',
+        required=True,
+        nargs='+',
+        help='Filter issues that belong to any of the given signatures.',
+    )
+    filter_group.add_argument(
+        '-m', '--message',
+        nargs='*',
+        help='Filter issues if any of the given messages is in the signature '
+             'body. A message can be given as a plain text or in the format '
+             'of root_issue_key:message. Any root issue that is assigned to a '
+             'message will be linked to the issues that will be resolved by it'
+    )
+
+    dry_run_group = parser.add_argument_group(title='Dry run options')
+    dry_run_args = dry_run_group.add_mutually_exclusive_group()
+    dry_run_args.add_argument(
+        '-d', '--dry-run',
+        action='store_true',
+        help='Dry run. Don\'t update tickets. Write output to stdout'
+    )
+    dry_run_args.add_argument(
+        '-t', '--dry-run-temp',
+        action='store_true',
+        help=f'Dry run. Don\'t update tickets. Write output to a temp file'
+    )
+
+    return parser.parse_args()
+
+
+class IssueData:
+
+    def __init__(self, issue, signature, comment):
+        self.issue = issue
+        self.signature = signature
+        self.comment = comment
+        self.root_issue = None
+
+
+def get_signatures(signature_types, jira):
+    signatures = filter(
+        lambda x: x.__name__.lower() in signature_types,
+        SIGNATURES,
+    )
+    return [sign(jira) for sign in signatures]
```

Comment:
Can't seem to find the definition of the `sign` function anywhere

Reply:
@RazRegev rename to ``signature``